tagstudio/src/qt/modals/drop_import.py

```diff
@@ -0,0 +1,150 @@
+from pathlib import Path
+import shutil
+import typing
+import logging
+
+from PySide6.QtGui import QDropEvent, QDragEnterEvent, QDragMoveEvent
+from PySide6.QtWidgets import QMessageBox
+
+if typing.TYPE_CHECKING:
+    from src.qt.ts_qt import QtDriver
+
+logging.basicConfig(format="[DROP IMPORT] %(message)s", level=logging.INFO)
+
+
+def dropEvent(driver: "QtDriver", event: QDropEvent):
+    if not event.mimeData().hasUrls():
+        return
+
+    import_files(driver, event.mimeData().urls())
+
+
+def dragEnterEvent(event: QDragEnterEvent):
+    if event.mimeData().hasUrls():
+        event.accept()
+    else:
+        event.ignore()
+
+
+def dragMoveEvent(event: QDragMoveEvent):
+    if event.mimeData().hasUrls():
+        event.accept()
+    else:
+        event.ignore()
+
+
+def import_files(driver: "QtDriver", urls):
+    files: list[Path] = []
+    dirs_in_root: list[Path] = []
+    duplicate_files: list[Path] = []
+
+    for url in urls:
+        if url.isLocalFile():
```

Comment:
again me with the same suggestion - there's no `else:` condition, so the logic can be flipped to avoid the code indention:

```
if not url.isLocalFile():
    continue
```